src/tools/r2rdump/x86/GcInfo.cs

```diff
@@ -0,0 +1,194 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection.PortableExecutable;
+using System.Text;
+using System.Xml.Serialization;
+
+namespace R2RDump.x86
+{
+    public class GcInfo : BaseGcInfo
+    {
+        public InfoHdrSmall Header { get; set; }
+        public GcSlotTable SlotTable { get; set; }
+
+        public GcInfo() { }
+
+        public GcInfo(byte[] image, int offset, Machine machine, ushort majorVersion)
+        {
+            Offset = offset;
+
+            CodeLength = (int)NativeReader.DecodeUnsignedGc(image, ref offset);
+
+            Header = InfoHdrDecoder.DecodeHeader(image, ref offset, CodeLength);
+
+            SlotTable = new GcSlotTable(image, Header, ref offset);
+
+            Transitions = new Dictionary<int, BaseGcTransition>();
+            if (Header.Interruptible)
+            {
+                GetTransitionsFullyInterruptible(image, ref offset);
+            }
+            else if (Header.EbpFrame)
```

Comment:
It doesn't look like any work is done in the other branches of the if statement. Can you remove those?